app/models/user.rb

```diff
@@ -99,6 +100,7 @@ def kind
     return "admin" if has_role?(Role::ORG_ADMIN, organization)
     return "normal" if has_role?(Role::ORG_USER, organization)
     return "partner" if has_role?(Role::PARTNER, partner)
+    return "deactived" if has_role?(Role::DEACTIVATED)
```

Comment:
This doesn't seem like the right way to model the user. We shouldn't be deliberately giving them a deactivated role. If the user has **no** roles, we should consider that user deactivated and treat them accordingly.

Reply:
Hrm.  @dorner -- Here's the thing, though.   The bank can deactivate a users and reactivate them.  However, there is also a case where someone may go to work for a partner after having worked for a bank (or vice versa), with possible overlap.   (pls check the issue ).   As things are now (deactivating the user as a user),  when the bank deactivates a user, it precludes them working for a partner.

Reply:
There may well be a way of allowing this situation without using a deactivated role,  *and* if we were to use a deactivated role (which, I admit, is how I thought this would be done, given that we allow reactivation, and that someone might (also) be working with a partner) , it would need to be deactivated in the context of an organization, not just deactivated,  imo.

Reply:
Ah - I wasn't aware that the bank needed to be able to reactivate users. If that's the case, then yes an org-level deactivated role probably makes the most sense. However, there might be code lurking around that assumes that if you have any role with an org, you have the most basic one (which is ORG\_USER) - so you might need to tweak that assumption.

Reply:
See RolifyCommunity/rolify#551 - similar idea but here we'd use `deactivated` instead of `deleted`.

Reply:
FWIW I have a gut "that sounds right" reaction to the idea of adding the "deactivated" column to the user_roles table.